cirq-core/cirq/vis/circuit_to_latex_quantikz.py

```diff
@@ -0,0 +1,710 @@
+# -*- coding: utf-8 -*-
+r"""Converts Cirq circuits to Quantikz LaTeX (using modern quantikz syntax).
+
+This module provides a class, `CirqToQuantikz`, to translate `cirq.Circuit`
+objects into LaTeX code using the `quantikz` package. It aims to offer
+flexible customization for gate styles, wire labels, and circuit folding.
+
+Example:
+    >>> import cirq
+    >>> from cirq.vis.circuit_to_latex_quantikz import CirqToQuantikz
+    >>> q0, q1 = cirq.LineQubit.range(2)
+    >>> circuit = cirq.Circuit(
+    ...     cirq.H(q0),
+    ...     cirq.CNOT(q0, q1),
+    ...     cirq.measure(q0, key='m0'),
+    ...     cirq.Rx(rads=0.5).on(q1)
+    ... )
+    >>> converter = CirqToQuantikz(circuit, fold_at=2)
+    >>> latex_code = converter.generate_latex_document()
+    >>> print(latex_code) # doctest: +SKIP
+    \documentclass[preview, border=2pt]{standalone}
+    % Core drawing packages
+    \usepackage{tikz}
+    \usetikzlibrary{quantikz} % Loads the quantikz library (latest installed version)
+    % Optional useful TikZ libraries
+    \usetikzlibrary{fit, arrows.meta, decorations.pathreplacing, calligraphy}
+    % Font encoding and common math packages
+    \usepackage[T1]{fontenc}
+    \usepackage{amsmath}
+    \usepackage{amsfonts}
+    \usepackage{amssymb}
+    % \usepackage{physics} % Removed
+    % --- Custom Preamble Injection Point ---
+    % --- End Custom Preamble ---
+    \begin{document}
+    \begin{quantikz}
+    \lstick{$q_{0}$} & \gate[style={fill=yellow!20}]{H} & \ctrl{1} & \meter{$m0$} \\
+    \lstick{$q_{1}$} & \qw & \targ{} & \qw
+    \end{quantikz}
+
+    \vspace{1em}
+
+    \begin{quantikz}
+    \lstick{$q_{0}$} & \qw & \rstick{$q_{0}$} \\
+    \lstick{$q_{1}$} & \gate[style={fill=green!20}]{R_{X}(0.5)} & \rstick{$q_{1}$}
+    \end{quantikz}
+    \end{document}
+"""
+# mypy: ignore-errors
+
+
+import math
+import warnings
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Type
+
+import cirq
```

Comment:
You cannot import cirq from within cirq.  
Please change this to: "from cirq import ops" and then change cirq.XPowGate to ops.XPowGate (etc).